File: BasicAlgorithms/Sorting/L1-insertion-sort/insertion-sort.py
my_array3 = [45, 10, 63, 40, 50, 80]

def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >=0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j+1] = key
    return arr

print(insertion_sort(my_array3))


# 45, 10, 63, 40, 50, 80
# 45, 45, 63, 40, 50, 80
#10, 45, 63, 40, 50, 80
#10, 45, 

File: BasicAlgorithms/Sorting/L1-insertion-sort/insertion-sort.js
let my_array3 = [45, 10, 63, 40, 50, 80]

const insertionSort = (arr)=>{
    for (let i = 1; i < arr.length; i++) {
        let key = arr[i]
        let j = i - 1
        while ( j >= 0 && arr[j] > arr[i]) {
            arr[j + 1] = arr[i]
            j -= 1
        }

        arr[j + 1] = key
    }

    return arr
}

console.log(insertionSort(my_array3))